src/EA.Weee.RequestHandlers/Admin/AatfReports/GetNonObligatedWeeeReceivedAtAatfsDataCsvHandler.cs

```diff
@@ -0,0 +1,87 @@
+﻿namespace EA.Weee.RequestHandlers.Admin.AatfReports
+{
+    using System;
+    using System.Threading.Tasks;
+    using Core.Admin;
+    using Core.Shared;
+    using DataAccess;
+    using DataAccess.StoredProcedure;
+    using Domain.Lookup;
+    using EA.Prsd.Core;
+    using EA.Weee.Requests.Admin.AatfReports;
+    using Prsd.Core.Mediator;
+    using Requests.Admin.AatfReports;
+    using Security;
+    using Shared;
+
+    internal class GetNonObligatedWeeeReceivedDataAtAatfsCsvHandler : IRequestHandler<GetUkNonObligatedWeeeReceivedAtAatfsDataCsv, CSVFileData>
+    {
+        private readonly IWeeeAuthorization authorization;
+        private readonly WeeeContext context;
+        private readonly CsvWriterFactory csvWriterFactory;
+        private readonly ICommonDataAccess commonDataAccess;
+
+        public GetNonObligatedWeeeReceivedDataAtAatfsCsvHandler(IWeeeAuthorization authorization, WeeeContext context,
+            CsvWriterFactory csvWriterFactory, 
+            ICommonDataAccess commonDataAccess)
+        {
+            this.authorization = authorization;
+            this.context = context;
+            this.csvWriterFactory = csvWriterFactory;
+            this.commonDataAccess = commonDataAccess;
+        }
+
+        public async Task<CSVFileData> HandleAsync(GetUkNonObligatedWeeeReceivedAtAatfsDataCsv request)
+        {
+            authorization.EnsureCanAccessInternalArea();
+
+            if (request.ComplianceYear == 0)
+            {
+                var message = $"Compliance year cannot be \"{request.ComplianceYear}\".";
+                throw new ArgumentException(message);
+            }
+
+            var items = await context.StoredProcedures.GetNonObligatedWeeeReceivedAtAatfs(request.ComplianceYear, request.AuthorityId, request.PatAreaId, request.AatfName);
+
+            var csvWriter = csvWriterFactory.Create<NonObligatedWeeeReceivedAtAatfsData>();
+            csvWriter.DefineColumn(@"Authority", i => i.Authority);
+            csvWriter.DefineColumn(@"Pat Area", i => i.PatArea);
+            csvWriter.DefineColumn(@"Area", i => i.Area);
+            csvWriter.DefineColumn(@"Year", i => i.Year);
+            csvWriter.DefineColumn(@"Quarter", i => i.Quarter);
+            csvWriter.DefineColumn(@"Submitted by", i => i.SubmittedBy);
+            csvWriter.DefineColumn(@"Date submitted", i => i.SubmittedDate);
+            csvWriter.DefineColumn(@"Organisation name", i => i.OrganisationName);
+            csvWriter.DefineColumn(@"Category", i => i.Category);
+            csvWriter.DefineColumn(@"Total non-obligated WEEE received (t)", i => i.TotalNonObligatedWeeeReceived);
+            csvWriter.DefineColumn(@"Non-obligated WEEE kept / retained by DCFs (t)", i => i.TotalNonObligatedWeeeReceivedFromDcf);
+            var fileContent = csvWriter.Write(items);
```

Comment:
Column names needs to be updated like Date submitted (GMT), WROS Pan area team

Reply:
Good spot...

Reply:
> Column names needs to be updated like Date submitted (GMT), WROS Pan area team

Good spot...

Reply:
Updated as removed the authority, area and wros area parameters